plutus-exe/src/Main.hs

```diff
@@ -179,11 +189,21 @@ availableExamples =
     , ("churchSucc" , SomeTermExample $ toTermExample PLC.churchSucc)
     ]
 
+getAvailableExamples :: IO [(ExampleName, SomeExample)]
+getAvailableExamples = do
+    interesting <- getInteresting
+    pure $ simpleExamples ++ map (second $ SomeTermExample . toTermExample) interesting
+
+-- The implementation is a little hacky: we generate interesting examples when the list of examples
+-- is requsted and at each lookup of a particular example. I.e. each time we generate distinct
+-- terms. But types of those terms must not change across requests, so we're safe.
 runExample :: ExampleOptions -> IO ()
-runExample (ExampleOptions ExampleAvailable)     =
-    traverse_ (T.putStrLn . PLC.docText . uncurry prettySignature) availableExamples
-runExample (ExampleOptions (ExampleSingle name)) =
-    T.putStrLn $ case lookup name availableExamples of
+runExample (ExampleOptions ExampleAvailable)     = do
+    examples <- getAvailableExamples
+    traverse_ (T.putStrLn . PLC.docText . uncurry prettySignature) examples
+runExample (ExampleOptions (ExampleSingle name)) = do
+    examples <- getAvailableExamples
```

Comment:
duplicated

Reply:
Yes, but is it a problem? I can factor out this line, but I do not feel like it would make anything nicer or easier to read.

Reply:
Oh, I just can't read, I thought this was in the same block :man_facepalming: